test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionProtocolTests.cs

```diff
@@ -39,6 +43,28 @@ public class HubConnectionProtocolTests
         }
 
         [Fact]
+        public async Task StreamSendsAnInvocationMessage()
+        {
+            var connection = new TestConnection();
+            var hubConnection = new HubConnection(connection, new JsonHubProtocol(new JsonSerializer()), new LoggerFactory());
+            try
+            {
+                await hubConnection.StartAsync();
+
+                var observable = hubConnection.Stream<object>("Foo");
+
+                var invokeMessage = await connection.ReadSentTextMessageAsync().OrTimeout();
+
+                Assert.Equal("{\"invocationId\":\"1\",\"type\":1,\"target\":\"Foo\",\"arguments\":[]}", invokeMessage);
```

Comment:
Because this is testing the HubProtocol logic. The only reason we "send" objects is because I added some small JSON serializing to the test code. I don't really think we want to JSON deserialize and then build a comparer.